File: Linkpoint/src/main/java/com/linkpoint/voice/common/messages/VoiceRejectCall.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  android.os.Bundle
 */
package com.linkpoint.voice.common.messages;

import android.os.Bundle;
import com.linkpoint.voice.common.VoicePluginMessage;
import com.linkpoint.voice.common.model.VoiceChannelInfo;
import javax.annotation.Nonnull;

public class VoiceRejectCall
implements VoicePluginMessage {
    @Nonnull
    public final String sessionHandle;
    public final VoiceChannelInfo voiceChannelInfo;

    public VoiceRejectCall(Bundle bundle) {
        this.sessionHandle = bundle.getString("sessionHandle");
        this.voiceChannelInfo = new VoiceChannelInfo(bundle.getBundle("voiceChannelInfo"));
    }

    public VoiceRejectCall(@Nonnull String string2, VoiceChannelInfo voiceChannelInfo) {
        this.sessionHandle = string2;
        this.voiceChannelInfo = voiceChannelInfo;
    }

    @Override
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putString("sessionHandle", this.sessionHandle);
        bundle.putBundle("voiceChannelInfo", this.voiceChannelInfo.toBundle());
        return bundle;
    }
}

File: Linkpoint/src/main/kotlin/com/linkpoint/voice/common/messages/VoiceRejectCall.kt
package com.linkpoint.voice.common.messages

import android.os.Bundle
import com.linkpoint.voice.common.VoicePluginMessage
import com.linkpoint.voice.common.model.VoiceChannelInfo
import javax.annotation.Nonnull

class VoiceRejectCall(
    val sessionHandle: String?,
    val voiceChannelInfo: VoiceChannelInfo
) : VoicePluginMessage {
    
    constructor(bundle: Bundle) : this(
        bundle.getString("sessionHandle"),
        VoiceChannelInfo(bundle.getBundle("voiceChannelInfo")!!)
    )

    override fun toBundle(): Bundle {
        val bundle = Bundle()
        bundle.putString("sessionHandle", sessionHandle)
        bundle.putBundle("voiceChannelInfo", voiceChannelInfo.toBundle())
        return bundle
    }
}